apps/emqx_bridge_tablestore/src/emqx_bridge_tablestore_connector.erl

```diff
@@ -0,0 +1,309 @@
+    case maps:find(ChannelId, Channels) of
+        {ok, #{table_name := TableName0, meta_update_model := MetaUpdateMode} = ChannelState} ->
+            try
+                Row = mk_tablestore_data_row(Message, ChannelState),
+                TableName = render_table_name(TableName0, Message),
+                mk_tablestore_data(TableName, MetaUpdateMode, [Row])
+            of
+                Data ->
+                    LogMsg = #{msg => ots_query, channel => ChannelId, data => Data},
+                    ?LOG_T(debug, LogMsg),
+                    case ots_ts_client:put(ClientRef, Data) of
+                        {ok, _Res} -> ok;
+                        {error, Reason} -> {error, {unrecoverable_error, Reason}}
+                    end
+            catch
+                throw:{bad_ots_data, _} = Reason ->
+                    {error, {unrecoverable_error, Reason}}
+            end;
+        error ->
+            {error, {unrecoverable_error, channel_not_found}}
+    end.
+
+on_batch_query(_, [{ChannelId, _} | _] = MsgList, #{client_ref := ClientRef, channels := Channels}) ->
+    case maps:find(ChannelId, Channels) of
+        {ok, ChannelState} ->
+            try mk_tablestore_batch_data(MsgList, ChannelState) of
+                BatchDataList ->
+                    send_batch_data(BatchDataList, ClientRef, ChannelId)
+            catch
+                throw:{bad_ots_data, _} = Reason ->
+                    {error, {unrecoverable_error, Reason}}
+            end;
+        error ->
+            {error, {unrecoverable_error, channel_not_found}}
+    end.
+
+send_batch_data(BatchDataList, ClientRef, ChannelId) ->
+    LogMsg = #{msg => ots_batch_query, channel => ChannelId, batch_data_list => BatchDataList},
+    ?LOG_T(debug, LogMsg),
+    Res = [ots_ts_client:put(ClientRef, BatchData) || BatchData <- BatchDataList],
+    Filter = fun
+        ({error, _}) -> true;
+        (_) -> false
+    end,
+    case lists:filter(Filter, Res) of
+        [] -> ok;
+        Errors -> {error, {unrecoverable_error, Errors}}
+    end.
+
+%%--------------------------------------------------------------------
+%% Internal Functions
+%%--------------------------------------------------------------------
+start_ots_ts_client(_, OtsOpts) ->
+    ots_ts_client:start(OtsOpts).
+
+list_ots_tables(ClientRef) ->
+    try
+        ots_ts_client:list_tables(ClientRef)
+    catch
+        Err:Reason:ST ->
+            {error, #{error => Err, reason => Reason, stacktrace => ST}}
+    end.
+
+preproc_tags(Tags) ->
+    maps:fold(
+        fun(K, V, AccIn) ->
+            AccIn#{maybe_preproc(bin(K)) => maybe_preproc(V)}
+        end,
+        #{},
+        Tags
+    ).
+
+preproc_fields(Fields) ->
+    lists:map(
+        fun(#{column := C, value := V} = Row0) ->
+            #{
+                column => maybe_preproc(C),
+                value => maybe_preproc(V),
+                isint => maybe_preproc(maps:get(isint, Row0, undefined)),
```

Comment:
why pre-process a boolean type config?

Reply:
The user may want to set this option by sending a MQTT payload such as:
 `{"column": "c", "value": 1, "is_int": true}`
 
So that he could write the config as:
`{column = "${payload.column}", value = "${payload.value}", isint = "${payload.is_int}"}`

Reply:
the config schema defined it as a boolean type.
https://github.com/emqx/emqx/pull/14414/files#diff-1eddbf46abf2a8500ad5ffa52984fd0a10bfcb5b943ff5b38a4fdb900c590c8cR258
it should be `union([boolean(), binary()])` ?

Reply:
fixed